### PR TITLE
Check RegisterMetricAndTrackRateLimiterUsage error when starting Tokens controller

### PR DESCRIPTION
**Pull request by frodenas**
**What this PR does / why we need it**:

Prevent `Tokens` controller to start if `metrics.RegisterMetricAndTrackRateLimiterUsage` returns an error.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: complements #53571

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -533,6 +533,9 @@ func (c serviceAccountTokenControllerStarter) startServiceAccountTokenController
 			RootCA:         rootCA,
 		},
 	)
+	if err != nil {
+		return true, fmt.Errorf("error creating service account token controller: %v", err)
```
Comment (liggitt) [inline cmd/kube-controller-manager/app/controllermanager.go]: Why return true here?

Comment (frodenas) [reply cmd/kube-controller-manager/app/controllermanager.go]: Consistency with other returns: [L511](https://github.com/frodenas/kubernetes/blob/ec31947cd88d709da15f6c618e8b698693ba9ca2/cmd/kube-controller-manager/app/controllermanager.go#L511), [L518](https://github.com/frodenas/kubernetes/blob/ec31947cd88d709da15f6c618e8b698693ba9ca2/cmd/kube-controller-manager/app/controllermanager.go#L518), [L521](https://github.com/frodenas/kubernetes/blob/ec31947cd88d709da15f6c618e8b698693ba9ca2/cmd/kube-controller-manager/app/controllermanager.go#L521).

**Comment by frodenas** [comment]
integration tests will fail until #53834 is merged.

/hold

**Comment by frodenas** [comment]
Blocked by #53935

**Comment by frodenas** [comment]
Superseded by #53592

**Pull request closed without merging**